harness/determined/profiler_design_april_2021.md

```diff
@@ -0,0 +1,114 @@
+# Profiler Design Notes (April 2021)
```

Comment:
Is the plan to commit this into the codebase?

Reply:
I see your comment now.  I'm not opposed to inline developer docs on principle (I've written [some](https://github.com/determined-ai/determined/blob/master/harness/determined/pytorch/README.md) myself).

But in this case I don't think the profiling system here is hard to understand, or convoluted, or presents a complex API to justify committing the docs to the repository.  Honestly I think the docstring that is already in ProfilingAgent's sums the system up plenty well.

Reply:
Removed it.